### PR TITLE
feat: add Herbarium to create REST endpoints

### PR DESCRIPTION
**Pull request by rikdprogramador**
Routes were created using custom files and repetitive writing. With this feature, there will be a
faster alternative to do this same route path as herbs2rest will use herbarium to generate the
endpoints. In addition, a test code was implemented to validate this function.

Fixes [#22](https://github.com/herbsjs/herbs2rest/issues/22)

### Author/Contributor
- [X] If documentation is needed for this change, has that been included in this pull request
- [ ] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: src/generateRoutesWithHerbarium.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const generateRoutes = require('./generateRoutes')
+
+function findEntitiesAndGroups(herbarium) {
+    const items = Array.from(herbarium.usecases.all.values()).map(e => ({ id: e.entity, group: e.group }))
```
Comment (italojs) [inline src/generateRoutesWithHerbarium.js:4]: could you use explicit name declaration in `e` variable?
following the clean code, in this way we could to build a easy code to read

**Review by italojs — CHANGES_REQUESTED** [review]

**Comment by italojs** [comment]
before we merge it, could you add an herbrarium session into https://herbsjs.org/docs/glues/herbs2rest, please?

**Comment by italojs** [comment]
great https://github.com/herbsjs/herbsjs.github.io/pull/117

**Comment by italojsv** [comment]
@dalssoft  @jhomarolo some one could to merge it?

**Comment by dalssoft** [comment]
I need to do make a deeper review for this PR but I haven't got the time. The way it is now it does not support simple cases, like overriding a route (which would be possible if the file for conversion was on the project, not in the lib).

**Comment by maikvortx** [comment]
@dalssoft What do you think about sending new PRs with new functionality like your example? This PR is just the beginning of the functionality with its basic purpose.

**Review by italojs — APPROVED** [review]

**Comment by jhomarolo** [comment]
@italojs it seems that @dalssoft requested some changes in his last comment

**Comment by dalssoft** [comment]
What if instead of giving `generateRoutesWithHebarium` to the developer (a function with (1) to many responsibilities and (2) ugly name) we just expose a new `getControllers`:

The developer would need to do just that on the application side:

```js
const { getControllers , generateRoutes } = require('@herbsjs/herbs2rest')

const controllers = getControllers(herbarium)
... // opportunity to change controllers object if needed
generateRoutes(controllers, routes, showEndpoints)
``` 

This also would give the oportunity to change the returned controllers object if needed.

File: package.json [hunk 2]
```diff
@@ -58,5 +58,8 @@
   },
   "publishConfig": {
     "access": "public"
+  },
+  "dependencies": {
```
Comment (dalssoft) [inline package.json:62]: It seems that `"@herbsjs/herbarium"` should be a `peerDependencies`, right?

**Review by dalssoft — CHANGES_REQUESTED** [review]

**Comment by italojs** [comment]
I dont agree the idea the herbrarium is a peer dependency as this lib still is in beta, we will change it a lot and the herbs2rest will need to be compatible with the developer herbrarium version, I'm not sure if this is a good idea

**Comment by dalssoft** [comment]
> herbs2rest will need to be compatible with the developer herbrarium version

since this feature is focused on creating a link between herbrarium and herbs2rest, I don't see how this can be accomplish it without the dependency

**Comment by italojs** [comment]
if we use it as a dependecy instead peerDependeecy when someone install the herbs2rest, the correct version for herbs2rest will be installed, so if the developer would like to install the herbrarium with a different version to use it in they project, they could to do it

**Comment by dalssoft** [comment]
> if we use it as a dependecy instead peerDependeecy when someone install the herbs2rest, the correct version for herbs2rest will be installed, so if the developer would like to install the herbrarium with a different version to use it in they project, they could to do it

however the herbrarium object instance passed to herbs2rest function will always be the one from the developer project, not from the herbs2rest, so herbs2rest will use the "host" version anyways

**Comment by italojs** [comment]
make sense

**Comment by jhomarolo** [comment]
@rikdprogramador do you still here in this PR?

**Comment by dalssoft** [comment]
Pls, check this comment: https://github.com/herbsjs/herbs2rest/pull/25#issuecomment-1100728874

**Comment by rikdprogramador** [comment]
I am working on it!

**Comment by jhomarolo** [comment]
This pr was resolved with #35

**Pull request closed without merging**